File: src/main/java/com/example/hbs/psi/impl/HbCloseBlockMustacheImpl.java
package com.example.hbs.psi.impl;

import com.example.hbs.psi.HbCloseBlockMustache;
import com.example.hbs.psi.HbOpenBlockMustache;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

public class HbCloseBlockMustacheImpl extends HbBlockMustacheImpl implements HbCloseBlockMustache {
  public HbCloseBlockMustacheImpl(@NotNull ASTNode astNode) {
    super(astNode);
  }

  @Override
  public HbOpenBlockMustache getPairedElement() {
    PsiElement openBlockElement = getParent().getFirstChild();
    if (openBlockElement instanceof HbOpenBlockMustache) {
      return (HbOpenBlockMustache)openBlockElement;
    }

    return null;
  }
}

File: src/main/kotlin/com/example/hbs/psi/impl/HbCloseBlockMustacheImpl.kt
package com.example.hbs.psi.impl

import com.example.hbs.psi.HbCloseBlockMustache
import com.example.hbs.psi.HbOpenBlockMustache
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement

class HbCloseBlockMustacheImpl(astNode: ASTNode) :
    HbBlockMustacheImpl(astNode), HbCloseBlockMustache {

    override fun getPairedElement(): HbOpenBlockMustache? {
        val openBlockElement: PsiElement? = parent.firstChild
        return openBlockElement as? HbOpenBlockMustache
    }
}
